### ISSUE TITLE
Call to a member function fieldtype() on null

### ISSUE THREAD

**Issue by karakhanyans** [statamic/seo-pro#238]
Hello. 

After upgrading to statamic v3.3 we started to see Call to a member function fieldtype() on null when we create a new collection item. However if during creation we disable SEO and save the item, it works fine, then we are able to edit that item and enable SEO, and everything starts to work for that item.

Can you please suggest a solution for this?

Thanks.

```
{
    "message": "Call to a member function fieldtype() on null",
    "exception": "Error",
    "file": "/home/forge/www-qa.felmo.de/releases/20221118131311/vendor/statamic/seo-pro/src/Fieldtypes/SourceFieldtype.php",
    "line": 109,
}
```

**Comment by jackmcdade** [comment]
What do you mean by disable SEO?

**Comment by karakhanyans** [comment]
@jackmcdade I mean when we disable it from SEO tab for the current item

<img width="409" alt="CleanShot 2022-11-21 at 17 14 49@2x" src="https://user-images.githubusercontent.com/12999189/203064538-aa89d70e-71c0-4a24-b009-e2745bc1dc3d.png">

**Comment by jasonvarga** [comment]
Would you be able to zip up the site and send it to support@statamic.com or provide us with GitHub access? Or even better, an example repo with the problem reproduced. I wasn't able to reproduce this here.

**Comment by karakhanyans** [comment]
@jasonvarga will it be possible to set up a call so I can show you the error via screen sharing? We can check the bug together.

**Comment by jasonvarga** [comment]
No but you could try recording a screencast for us. https://zipmessage.com/statamic

It would be much better if you could reproduce the bug and share a repo with us, though.

**Comment by karakhanyans** [comment]
Just shared the screen recording: https://zipmessage.com/statamic/conversation-with-sergey

**Comment by karakhanyans** [comment]
Hi @jasonvarga do you have any news regarding my last screen recording? 

Thanks.

**Comment by jesseleite** [comment]
@karakhanyans The screencast is good, but we are having trouble reproducing the issue here. Can you paste your collection's yaml config file? (Specifically the `inject` part with your `seo` config, which should be located in `content/collections/your-collection-handle.yaml`.)

> After upgrading to statamic v3.3 

^ Also were you migrating this site from Statamic v2 by chance?

**Comment by karakhanyans** [comment]
Thanks for your response, @jesseleite 

We have upgraded from Statamic v3.2

As we are unable to create new collection item, sending you one existing. And for existing ones we don't have any troubles updating them, problem is with creating new item with SEO enabled.

```
blueprint: districts
district_name: Charlottenburg-Wilmersdorf
title: Adlershof
hero_title: 'Hausbesuche vom Tierarzt - Mobiler Tierarzt Adlershof'
districts_title_heading: h3
quarter_area: '10,6'
quarter_inhabitants: '130.223'
amount_cats: '12.154'
amount_dogs: '5.139'
city: berlin
seo:
  title: 'Mobiler Tierarzt Charlottenburg, Berlin'
  description: 'felmo ist der mobile Tierarzt in Charlottenburg ✓ Stressfrei für Deinen Liebling ✓ Einfach online freien Termin buchen.'
template: cities/district
```

**Comment by jesseleite** [comment]
@karakhanyans I didn't mean an entry, I meant your collection's config (again, located at `content/collections/your-collection-handle.yaml`).

**Comment by karakhanyans** [comment]
@jesseleite Sorry, I misunderstood you, here is it

```
title: Districts
template: cities/district
layout: layout
revisions: false
route: '/tierarzt/new/{city}/{slug}'
sort_dir: asc
columns:
  - title
  - slug
inject:
  seo:
    title: 'Tierarzt Berlin {{title}} · felmo mobiler Tierarzt 😸🐶'
    description: 'Kostenlose Anfahrt nach {{title}} · Stressfreie Hausbesuche für Hunde & Katzen · Preise und Behandlungen wie in der Tierarztpraxis · Termin buchen'
    sitemap: true
    show_future: true
    show_past: true

```

**Comment by jesseleite** [comment]
Judging by that `seo` config there, it looks like you had migrated from v2 at one point using statamic/migrator?

Also, can you try removing that `sitemap: true` line and let me know if it fixes your problem with saving entries?

**Comment by karakhanyans** [comment]
@jesseleite thank you very much! Removing `sitemap: true` worked for us!

**Comment by jesseleite** [comment]
@karakhanyans Can you confirm that you at one point had migrated from v2 using statamic/migrator? Trying to track down how that got there in the first place, and I believe that's where it's happening.

**Comment by karakhanyans** [comment]
@jesseleite We've migrated from v2 to v3, but we did it manually and not used statamic/migrator.

**Comment by jesseleite** [comment]
@karakhanyans Fair enough! We fixed this in 3.1.2 anyway, in case a bad migration affects other users 👍
